examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -627,10 +628,10 @@ func (g *genDeepCopy) doStruct(t *types.Type, sw *generator.SnippetWriter) {
 			} else {
 				// Fall back on the slow-path and hope it works.
 				// TODO: don't depend on kubernetes code for this
-				sw.Do("if newVal, err := c.DeepCopy(&in.$.name$); err != nil {\n", args)
-				sw.Do("return err\n", nil)
-				sw.Do("} else {\n", nil)
+				sw.Do("if newVal, err := c.DeepCopy(&in.$.name$); err == nil {\n", args)
```

Comment:
Ah, I see that this is creating locally scoped errors.  This is still ugly, but I guess I can live with it.